lib/rules/maximum-line-length.js

```diff
@@ -58,9 +63,28 @@ module.exports.prototype = {
                 this._tabSize += ' ';
             }
 
-            this._allowRegex = (maximumLineLength.allowRegex === true);
-            this._allowComments = (maximumLineLength.allowComments === true);
-            this._allowUrlComments = (maximumLineLength.allowUrlComments === true);
+            var exceptions = maximumLineLength.ignore || [];
+            this._allowRegex = (exceptions.indexOf('regex') !== -1);
+            this._allowComments = (exceptions.indexOf('comments') !== -1);
+            this._allowUrlComments = (exceptions.indexOf('urlComments') !== -1);
+            this._allowFunctionSignature = (exceptions.indexOf('functionSignature') !== -1);
+
+            if (maximumLineLength.hasOwnProperty('allowRegex')) {
+                console.log(DEPRECATED.replace('${option}', this.getOptionName())
```

Comment:
The thing is, output might not be terminal, but anything user want, this is why we have reporters.

So only mentioning them in the docs as deprecated is fine.